src/Writers/LaTeXWriter.jl

```diff
@@ -83,23 +105,63 @@ function render(doc::Documents.Document)
-                pdf = replace("$(doc.user.sitename).pdf", " " => "")
-                try
-                    run(`latexmk -f -interaction=nonstopmode -view=none -lualatex -shell-escape $file`)
-                catch err
-                    @warn "failed to compile. Check generated LaTeX file."
-                    cp(file, joinpath(outdir, file); force = true)
-                end
-                cp(pdf, joinpath(outdir, pdf); force = true)
-            else
-                @warn "`latexmk` and `lualatex` required for PDF generation."
-            end
+
+            # compile .tex and copy over the .pdf file if compile_tex return true
+            status = compile_tex(doc, settings, texfile)
+            status && cp(pdffile, joinpath(doc.user.root, doc.user.build, pdffile); force = true)
         end
     end
 end
 
+const DOCKER_IMAGE_TAG = "0.1"
+
+function compile_tex(doc::Documents.Document, settings::LaTeX, texfile::String)
+    if settings.engine == "latexmk"
+        Sys.which("latexmk") === nothing && (@error "LaTeXWriter: latexmk command not found."; return false)
+        @info "LaTeXWriter: using latexmk to compile tex."
+        try
+            piperun(`latexmk -f -interaction=nonstopmode -view=none -lualatex -shell-escape $texfile`)
+            return true
+        catch err
+            logs = cp(pwd(), mktempdir(); force=true)
+            @error "LaTeXWriter: failed to compile tex with latexmk. " *
+                   "Logs and partial output can be found in $(Utilities.locrepr(logs))." exception = err
+            return false
+        end
+    elseif settings.engine == "docker"
+        Sys.which("docker") === nothing && (@error "LaTeXWriter: docker command not found."; return false)
+        @info "LaTeXWriter: using docker to compile tex."
+        script = """
+            mkdir /home/zeptodoctor/build
+            cd /home/zeptodoctor/build
+            cp -r /mnt/. .
+            latexmk -f -interaction=nonstopmode -view=none -lualatex -shell-escape $texfile
+            """
+        try
+            piperun(`docker run -itd -u zeptodoctor --name latex-container -v $(pwd()):/mnt/ --rm juliadocs/documenter-latex:$(DOCKER_IMAGE_TAG)`)
+            piperun(`docker exec -u zeptodoctor latex-container bash -c $(script)`)
+            piperun(`docker cp latex-container:/home/zeptodoctor/build/. .`)
+            return true
+        catch err
+            logs = cp(pwd(), mktempdir(); force=true)
+            @error "LaTeXWriter: failed to compile tex with docker. " *
+                   "Logs and partial output can be found in $(Utilities.locrepr(logs))." exception = err
+            return false
+        finally
+            try; piperun(`docker stop latex-container`); catch; end
+        end
+    else
+        @error "LaTeXWriter: unrecognized engine: $(engine)"
+        return false
+    end
+end
+
+function piperun(cmd)
+    verbose = "--verbose" in ARGS
```

Comment:
Is this ok? running with verbose as default for the tex stuff is a pain since it results in huge amounts of output...

Reply:
Maybe do it with an environment variable instead? That way you can also easily get the log on Travis.

Reply:
Added `DOCUMENTER_VERBOSE`